src/shared/bpf-dlopen.c

```diff
@@ -205,3 +206,20 @@ int dlopen_bpf(void) {
                                "libbpf support is not compiled in.");
 }
 #endif
+
+bool cgroup_bpf_supported(void) {
+        static int supported = -1;
+        int r;
+
+        if (supported >= 0)
+                return supported;
+
+        r = dlopen_bpf();
```

Comment:
So, hmm, this is effectively dlopen_bpf_cached. Note that dlopen_many_sym_or_warn() already becomes nop if dlp is initialized, hence dlopen_bpf() should just follow suit.